### PR TITLE
OCC-29: Add publish version input to publish-nuget.

### PR DESCRIPTION
**Pull request by sarahelsaig**
[OCC-29](https://lombiq.atlassian.net/browse/OCC-29)

### CODE REVIEW

**Review by DemeSzabolcs — APPROVED** [review]

**Comment by DemeSzabolcs** [comment]
Could someone else review this too, please? I'm not that home yet in GitHub actions, and I don't want to miss something. @0liver @Piedone

File: .github/actions/publish-nuget/action.yml [hunk 1]
```diff
@@ -37,8 +44,24 @@ runs:
...
+
+        if ($version -eq 'USE_GITHUB_REF_NAME') { $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v") }
+        if ($version -eq 'USE_GITHUB_RUN_NUMBER') {
+          $tags = git tag | 
+            ? { $_ -match '^v?\d+(\.\d+){0,3}$' } | 
+            % { [version]($_.TrimStart("v")) } | 
+            Sort-Object -Descending
+          $current = if ($tags.Count -gt 0) { $tags[0] } else { [version]"0.0.0" }
+
+          $next = New-Object "System.Version" $current.Major, $current.Minor, ($current.Build + 1)
```
Comment (Piedone) [inline .github/actions/publish-nuget/action.yml:60]: Why is this `$current.Build + 1`, not `$current.Build`?

**Comment by Piedone** [comment]
I only have the above remark, will approve if I'm OK with it.

File: .github/actions/publish-nuget/action.yml (hunk 1 above)
Comment (sarahelsaig) [reply .github/actions/publish-nuget/action.yml:60]: It's based on what's written in the wiki [here](https://lombiq.atlassian.net/wiki/spaces/DEV/pages/786857987/Managing+releases+of+open-source+projects#:~:text=then%20always%20use%20the%20next%20patch%20version). The idea is to create a new pre-release version after the current latest release with a strictly monotonically increasing pre-release suffix. The algorithm is:
- find what's the current latest release using the tags (excluding the ones with pre-release suffix)
  - if none found, use "0.0.0".
- increment the Build version (e.g. 4.6.0 => 4.6.1 or 0.0.0 => 0.0.1)
- add the `-preview-{GITHUB_RUN_NUMBER}` suffix.

Comment (Piedone) [reply .github/actions/publish-nuget/action.yml:60]: Ah I see, OK then.

**Review by Piedone — APPROVED** [review]

**Pull request merged**